numpy/core/src/multiarray/datetime.c

```diff
@@ -1761,6 +1762,9 @@ parse_datetime_unit_from_string(char *str, Py_ssize_t len, char *metastr)
                 return NPY_FR_as;
         }
     }
+    else if (allow_generic && !strcmp(str, "generic")) {
```

Comment:
Also curious as to why "generic" is not allowed? Thinking a bit more, I think my suggestion is not correct if `len` is not 7.  What is supposed to be there for a valid generic time? But maybe
```
else if (allow_generic && !(len == 7 && strncmp(str, "generic", 7))
```